pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -287,6 +284,12 @@ func (c *EtcdProxyController) syncHandler(key string) error {
 		return nil
 	}
 
+	var errs []error
+	etcdstorageCondition := etcdstoragev1alpha1.EtcdStorageCondition{
```

Comment:
Errors are now aggregated. When something errors, we're appending it here, and then we continue to next phase. This way, we're trying to do as much as we can before failing.

For conditions, we're starting with Deployed Unknown, and then we're changing it as appropriate. The condition is updated only once, at the end.